### PR TITLE
Cmake fix

### PR DESCRIPTION
**Pull request by anujm1**
Couple of fixes to CmakeLists.txt when using Yocto to cross-compile where we pass DESTDIR to cmake.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -203,7 +203,7 @@ if(DNNL_GPU_RUNTIME STREQUAL "OCL")
     install(FILES
         "../cmake/FindOpenCL.cmake"
         DESTINATION ${LIB_CONFIG_INSTALL_DIR})
-    install(FILES "${CMAKE_INSTALL_PREFIX}/${LIB_CONFIG_INSTALL_DIR}/FindOpenCL.cmake"
```
Comment (rsdubtso) [inline src/CMakeLists.txt:206]: Can you please clarify why this change is needed? Is it because there is an extra prefix?

Comment (anujm1) [reply src/CMakeLists.txt:206]: Yes looks related. If you pass a DESTDIR to cmake_install.cmake, then lets say you are installing to: 
/usr/lib/cmake/ and DESTDIR is D, FindOpenCL.cmake actually went to D//usr/lib/cmake/. So the next step will fail when copying file from /usr/lib/cmake.

Comment (anujm1) [reply src/CMakeLists.txt:206]: I had tried to include comments in the commit message. Perhaps should have copied them in pull request too ...

Comment (rsdubtso) [reply src/CMakeLists.txt:206]: Thanks. I should have read the commit messages. How standard is `DEST`? Do you know if other packaging tools use it too? I would probably prefer a mechanism that all the packagers could use.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -137,7 +137,7 @@ install(EXPORT ${LIB_EXPORT_NAME}
 
 # Intel MKL-DNN compat cmake files
 install(CODE "execute_process(COMMAND ${CMAKE_COMMAND}
-    -DDIR=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake
+    -DDIR=\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake
```
Comment (emfomenk) [inline src/CMakeLists.txt:140]: There are other places where `${CMAKE_INSTALL_PREFIX}` is used. Should we replace all of them with `\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}`? If so, probably, a new variable is required.

Also, current approach requires trailing `/` in `$DESTDIR`. Maybe that should be handled gracefully.

Comment (emfomenk) [reply src/CMakeLists.txt:140]: Oh, I probably got it. Please discard the comment about `/`.

Is the thing that for all the other `install()` commands `DESTDIR` works out of the box, but this particular `install` requires `\$ENV{DESTDIR}` explicitly because it performs some side work and `cmake` cannot understand that it needs to append `$DESTDIR`?

File: src/CMakeLists.txt (hunk 1 above)
Comment (emfomenk) [reply src/CMakeLists.txt:206]: side note -- overall this change looks reasonable. why would we copy: `source->dst1, dst1->dst2` instead of `source->dst1,source->dst2`? so I am oK with the change too.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -137,7 +137,7 @@ install(EXPORT ${LIB_EXPORT_NAME}
 
 # Intel MKL-DNN compat cmake files
```
Comment (emfomenk) [inline src/CMakeLists.txt:138]: (random place) please change the commit messages -- start them with `build: ` instead of `src/CMakefile.txt`

File: src/CMakeLists.txt (hunk 1 above)
Comment (rsdubtso) [reply src/CMakeLists.txt:206]: > why would we copy: source->dst1, dst1->dst2 instead of source->dst1,source->dst2? 

Guilty. I actually thought that it would be simpler this way. I'm also OK with this change. I need to test if `DESTDIR` is supported by non-Makefile generators.

File: src/CMakeLists.txt (hunk 2 above)
Comment (anujm1) [reply src/CMakeLists.txt:140]: Yes. We don't need to use DESTDIR when the path is for a destination to be installed - cmake automatically resolves the correct path if DESTDIR is passed. In this case, it is important that cmake_install.cmake has $ENV{DESTDIR} so the absolute path can then be set to DIR which is then used by gen_mkldnn_compat_cmakes.cmake.

Comment (emfomenk) [reply src/CMakeLists.txt:140]: I see, Thanks for confirming this!
